chanzuckerberg/idseq-web
# [Data Discovery] Refactor search

* Split between client and server searches:
	- Client searches uses information from dimensions to feed the suggestions on the search box
	- Server searches, as before, triggers a server on server side (for models with a high count like samples and taxons)
* Refactor server searches to be able to restrain by domain (i.e. public, my_data, etc.)
* Created a new component LiveSearchBox (which is a refactor of SearchBox) that is more flexible on how search suggestions is loaded. Also solves some bugs on triggering concurrent searches and avoid using outdated results.
* Some UI styling adjustments to the search box results


Testing: 
* Search for each of the models that search box in the header supports (projects, samples, taxa, location, host and tissues) and verify if the correct filter or action is triggered.
* Switch tabs (public and my data) and verify that results are constrained to the corresponding domain.
* As a follow up of this change, I will work on a manual script to validate data discovery for more comprehensive testing.
